File: common/src/main/java/work/fking/pangya/common/server/ServerConfig.java
package work.fking.pangya.common.server;

import java.util.List;

public record ServerConfig(
        int id,
        String name,
        int capacity,
        String bindAddress,
        String advertiseAddress,
        int port,
        List<ServerFlag> flags,
        List<ServerBoost> boosts,
        ServerIcon icon
) {

}

File: common/src/main/kotlin/work/fking/pangya/common/server/ServerConfig.kt
package work.fking.pangya.common.server

@JvmRecord
data class ServerConfig(
    val id: Int,
    val name: String,
    val capacity: Int,
    val bindAddress: String,
    val advertiseAddress: String,
    val port: Int,
    val flags: List<ServerFlag> = emptyList(),
    val boosts: List<ServerBoost> = emptyList(),
    val icon: ServerIcon = ServerIcon.BLACK_PAPEL
)
